akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -159,6 +165,48 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
         )
       }
     }
+
+    "respect settings" should {
+      "received SETTINGS_MAX_CONCURRENT_STREAMS should limit the number of outgoing streams" in new TestSetup(
```

Comment:
We need another test when the limit is changed later. Especially, if the current number of streams is then already above the limit.